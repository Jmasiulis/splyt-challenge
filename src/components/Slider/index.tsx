import React, { useState } from 'react';
import MaterialSlider from '@material-ui/core/Slider';

import StyledWrapper from './styles';

interface SliderProps {
  min: number;
  max: number;
  defaultValue: number;
  onValueChange: (value: number | number[]) => void;
}

export default function Slider({
  min, max, defaultValue, onValueChange,
}: SliderProps): React.ReactElement {
  const [value, setValue] = useState(defaultValue);

  return (
    <StyledWrapper>
      <MaterialSlider
        value={value}
        defaultValue={defaultValue}
        aria-labelledby="discrete-slider-small-steps"
        step={1}
        marks
        min={min}
        max={max}
        valueLabelDisplay="auto"
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore
        onChange={(e, newValue): void => setValue(newValue)}
        onChangeCommitted={(e, newValue): void => onValueChange(newValue)}
      />
      <span>
        Selected number of drivers:
        {' '}
        {value}
      </span>
    </StyledWrapper>
  );
}
