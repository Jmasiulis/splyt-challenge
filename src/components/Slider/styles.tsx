import styled from 'styled-components';

const StyledWrapper = styled.div`
  margin: 40px 10px 10px 10px;
  width: 50%;
`;

export default StyledWrapper;
