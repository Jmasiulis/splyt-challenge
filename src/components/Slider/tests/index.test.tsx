/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';

import Slider from '..';

describe('<Slider />', () => {
  const defaultProps = {
    min: 1,
    max: 50,
    defaultValue: 10,
    onValueChange: jest.fn(),
  };

  it('should render with default props', () => {
    const wrapper = shallow(<Slider {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
