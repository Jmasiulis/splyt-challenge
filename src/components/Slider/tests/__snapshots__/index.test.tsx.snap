// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render with default props 1`] = `
<styled.div>
  <WithStyles(ForwardRef(Slider))
    aria-labelledby="discrete-slider-small-steps"
    defaultValue={10}
    marks={true}
    max={50}
    min={1}
    onChange={[Function]}
    onChangeCommitted={[Function]}
    step={1}
    value={10}
    valueLabelDisplay="auto"
  />
  <span>
    Selected number of drivers:
     
    10
  </span>
</styled.div>
`;
